nucypher/blockchain/eth/sol/source/contracts/MinersEscrow.sol

```diff
@@ -569,36 +636,36 @@ contract MinersEscrow is Issuer {
     }
 
     /**
-    * @notice Confirm activity for the next period
-    * @param _miner Miner
-    * @param _lockedValue Locked tokens in the next period
-    * @param _additional Additional locked tokens in the next period.
-    * Used only if the period has already been confirmed
-    * @param _lastActivePeriod Last active period
+    * @notice Confirm activity for the next period and mine for the previous period
     **/
-    function confirmActivity(
-        address _miner,
-        uint256 _lockedValue,
-        uint256 _additional,
-        uint16 _lastActivePeriod
-    ) internal {
-        require(_lockedValue > 0);
-        MinerInfo storage info = minerInfo[_miner];
+    function confirmActivity() external {
+        address miner = msg.sender;
```

Comment:
Isn't the worker who has to confirm activity? @szotov

Reply:
Two case are possible:
1. Worker confirms through some staking contract
2. Worker confirms without any intermediary steps